pypeit/core/wavecal/echelle.py

```diff
@@ -0,0 +1,213 @@
+
+
+import os
+import sys
+import numpy as np
+from astropy.io import fits
+from pypeit import msgs
+from pypeit.core import fitting
+from pypeit.core.wavecal import wvutils
+from scipy import interpolate
+from astropy import constants as const
+from astropy.table import Table
+from IPython import embed
+c_kms = const.c.to('km/s').value
+
+
+
+def predict_ech_order_coverage(angle_fits_params, xd_angle_coeffs, xdisp, xdangle, norders, pad=0):
+    """
+    Predict the coverage of orders in the echelle spectrum using the disperser dependent
+    fits of the reddest order as a function of xdangle.
+
+    Args:
+        angle_fits_params (astropy.table.Table):
+            Table holding the arxiv parameters
+        xd_angle_coeffs
+            Table holding the arxiv data
+        xdisp (str):
+            Corss disperser. For HIRES this is either 'UV' or 'RED'
+        xdangle (float):\
+            Cross-disperser angle.
+        norders (int):
+            Number of orders identified on the detector
+        pad (int):
+            Number of orders to pad the coverage by on the blue and red side.
+
+    Returns:
+        order_vec (numpy.ndarray):
+            Array of order numbers for the predicted coverage.
+    """
+
+    xd_min, xd_max = angle_fits_params['xd_xmin'], angle_fits_params['xd_xmax']
+    idisp = angle_fits_params['xdisp_vec'] == xdisp
+    reddest_order_fit = int(np.round(
+        fitting.evaluate_fit(xd_angle_coeffs[idisp, :].flatten(), angle_fits_params['xd_func'], xdangle,
+                             minx=xd_min, maxx=xd_max)))
+    order_vec = reddest_order_fit + (np.arange(norders + 2*pad) - pad)[::-1]
+
+    return order_vec
+
+def predict_ech_wave_soln(angle_fits_params, ech_angle_coeffs, ech_angle, order_vec, nspec):
+    """
+    Predict an echelle spectrum wavelength solution for each order by evluating the polynomial fits of
+    wavelength solution coefficients vs echelle angle at the given echelle angle.
+
+    Args:
+        angle_fits_params (astropy.table.Table):
+            Table holding the parameters governing the echelle angle fits
+        ech_angle_coeffs (numpy.ndarray):
+            Array holding the polynomial coefficients for the fits of the wavelength solution polynomial coefficients
+            vs echelle angle.
+        ech_angle (float):
+            Echelle angle
+        order_vec (numpy.ndarray):
+            Array of order numbers for the deisred predicted spectrum. Shape = (norders,)
+        nspec (int):
+            Number of spectral pixels in the echelle spectrum
+
+    Returns:
+        wave_soln_guess (numpy.ndarray):
+            Array containing the predicted echelle spectrum. Shape = (nspec, norders)
+
+    """
+
+    norders = order_vec.size
+    wave_soln_guess = np.zeros((nspec, norders))
+    n_final = angle_fits_params['ech_n_final']
+
+    order_min, order_max = angle_fits_params['order_min'], angle_fits_params['order_max']
```

Comment:
why unpack these?  makes it a bit harder to know where they came from down below

Reply:
Change made.